eng/common/cross/toolchain.cmake

```diff
@@ -71,8 +71,17 @@ endif()
 
 if(TARGET_ARCH_NAME MATCHES "^(arm|armel)$")
   add_compile_options(-mthumb)
-  add_compile_options(-mfpu=vfpv3)
-  if(TARGET_ARCH_NAME STREQUAL "armel")
+  if (NOT DEFINED ARM_FPU_TYPE)
+    set(ARM_FPU_TYPE vfpv3)
+  endif(ARM_FPU_TYPE)
+
+  add_compile_options(-mfpu=${ARM_FPU_TYPE})
+  if (NOT DEFINED ARM_FPU_CAPABILITY)
+    set(ARM_FPU_CAPABILITY 0x7)
+  endif(ARM_FPU_CAPABILITY)
+  add_compile_options(-DCLR_FPU_CAPABILITY=${ARM_FPU_CAPABILITY})
```

Comment:
I am wondering whether we really need `ARM_FPU_CAPABILITY`. Can we just infer it from `ARM_FPU_TYPE`?

Reply:
AFAIK, there are too many different combinations of VFP types from arm that we'd all have to enumerate each and every single combination here (messier IMO). ARM's original idea was to define these bitmasks with _ARM_FP and have application detect the FPU capability by looking at the half, single and double precision bits. 

This works for GCC and ARMCC. Unfortunately, clang is not compliant. I tested this the other day. We are sort of emulating ARM's idea with bitmask here.

Reply:
The other problem is that vfp types used for gcc and clang are not identical. you can use vfp3-d32 on gcc. You can't do this on clang for instance. I thought it is better to push this to whoever is doing the port specify what his platform supports.

Reply:
ok